relayer/pkg/chainlink/keys/ocr2key.go

```diff
@@ -16,25 +16,25 @@ import (
 	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2/types"
 )
 
-var _ ocrtypes.OnchainKeyring = &ocr2Keyring{}
+var _ ocrtypes.OnchainKeyring = &OCR2Keyring{}
 
-type ocr2Keyring struct {
+type OCR2Keyring struct {
 	privateKey starksig.PrivateKey
 }
 
-func newStarkNetKeyring(material io.Reader) (*ocr2Keyring, error) {
+func NewStarkNetKeyring(material io.Reader) (*OCR2Keyring, error) {
```

Comment:
nbd but curious if you considered returning the already public interface instead?
```suggestion
func NewStarkNetKeyring(material io.Reader) (ocrtypes.OnchainKeyring, error) {
```

Reply:
I know there are differing opinions on when you should return the underlying type vs. the interface, but FWIW this can be error prone by accidentally leading to a typed nil (`(*OCR2Keyring)(nil)`) being set in a `type ocrtypes.OnchainKeyring` var which pases nil chekcs unexpectedly.

Reply:
that's fair! at the moment, the `OCR2Keyring` also needs to export marshal and unmarshal functions which are not part of `ocrtypes.OnchainKeyring` interface

another option could be to wrap the `OnchainKeyring` interface?

Reply:
Yeah, sounds like that would work. Really nbd though :shrug: